manual/thread_safetiness/index.md

```diff
@@ -0,0 +1,23 @@
+---
+title: Thread safetiness
+layout: single
+sidebar:
+  nav: "manual"
+toc: true
+toc_sticky: true
+---
+
+ROOT classes support one of the following thread-safety levels:
+
+- Thread unsafe: objects must not be used concurrently by different threads, even if objects are not shared among threads.
+
+- Conditionally thread safe: objects can be used safely by multiple threads concurrently as long as mutiple threads do not share the same object (i.e. every thread uses a private copy of its objects).  In addition, objects can be shared among threads as long as all threads use only `const` methods of the shared objects.
+
+- Thread safe: objects can safely be shared and used concurrently by multiple threads.
+
+Unless otherwise stated, classes starting with `R` (e.g. `RDataFrame`) are conditionally thread safe.
+Classes starting with `T` (e.g. `TDirectory`) are thread unsafe.
```

Comment:
Actually TDirectory (and as far as I can all the 'core' I/O classes) are "conditionally thread safe".

To find an example of thread unsafe.  An example of thread unsafe class is TUrl since it has const method that modify its content (via mutable data members) without taking any locks.

Reply:
```suggestion
Classes starting with `T` (e.g. `TUrl`) should be considered thread unsafe unless otherwise stated.

...

The classes involved in the I/O (e.g. `TDirectory`) are generally conditionally thread safe.
The classes involved in the retrieving of global information (eg. `TROOT`, `TClass`) are generally thread safe.
Most ROOT collections (eg. `TList`, `TObjArray`) are conditionally thread safe and can be turned into thread safe collection by calling `UseRWLock()` (on each given instances).

```

Reply:
"generally" thread safe sounds scary

Reply:
Well ... english is hard :) the "generally" was meant to bind to the list of classes not the thread safetyness :).  I.e. I was not sure enough to say that "All" the classes involved in the retrieving global information were thread safe.